tests/scripts/thread-cert/network_diag.py

```diff
@@ -0,0 +1,336 @@
+#!/usr/bin/env python3
+#
+#  Copyright (c) 2020, The OpenThread Authors.
+#  All rights reserved.
+#
+#  Redistribution and use in source and binary forms, with or without
+#  modification, are permitted provided that the following conditions are met:
+#  1. Redistributions of source code must retain the above copyright
+#     notice, this list of conditions and the following disclaimer.
+#  2. Redistributions in binary form must reproduce the above copyright
+#     notice, this list of conditions and the following disclaimer in the
+#     documentation and/or other materials provided with the distribution.
+#  3. Neither the name of the copyright holder nor the
+#     names of its contributors may be used to endorse or promote products
+#     derived from this software without specific prior written permission.
+#
+#  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+#  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+#  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+#  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+#  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+#  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+#  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+#  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+#  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+#  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+#  POSSIBILITY OF SUCH DAMAGE.
+#
+
+import struct
+
+import common
+import ipaddress
+import mle
+import network_layer
+
+from enum import IntEnum
+
+
+class TlvType(IntEnum):
+    EXT_ADDRESS = 0
+    ADDRESS16 = 1
+    MODE = 2
+    POLLING_PERIOD = 3
+    CONNECTIVITY = 4
+    ROUTE64 = 5
+    LEADER_DATA = 6
+    NETWORK_DTA = 7
+    IPV6_ADDRESS_LIST = 8
+    MAC_COUNTERS = 9
+    BATTERY_LEVEL = 14
+    SUPPLY_VOLTAGE = 15
+    CHILD_TABLE = 16
+    CHANNEL_PAGES = 17
+    TYPE_LIST = 18
+    MAX_CHILD_TIMEOUT = 19
+
+
+class Ipv6AddressList(object):
```

Comment:
no need for inherit from `object`?
```suggestion
class Ipv6AddressList:
```